internal/jobs/launch_instance_aws.go

```diff
@@ -15,38 +15,58 @@ import (
 )
 
 type LaunchInstanceAWSTaskArgs struct {
-	AccountID     int64 `json:"account_id"`
-	PubkeyID      int64 `json:"pubkey_id"`
+	// Associated reservation
 	ReservationID int64 `json:"reservation_id"`
-	// Optional name, can be blank
-	Name string `json:"name"`
-	// AWS AMI
+
+	// Associated account
+	AccountID int64 `json:"account_id"`
+
+	// Associated public key
+	PubkeyID int64 `json:"pubkey_id"`
+
+	// Detail information
+	Detail *models.AWSDetail `json:"detail"`
+
+	// AWS AMI as fetched from image builder
 	AMI string `json:"ami"`
-	// Amount of instances to launch
-	Amount int32 `json:"amount"`
-	// Immediately power off the system
-	PowerOff bool `json:"poweroff"`
-	// Amazon EC2 Instance Type
-	InstanceType string `json:"instance_type"`
+
 	// The ARN fetched from Sources which is linked to a specific source
 	ARN string `json:"arn"`
 }
 
+// Unmarshall arguments and handle error
 func HandleLaunchInstanceAWS(ctx context.Context, job dejq.Job) error {
-	ctxLogger := ctxval.Logger(ctx)
-	ctxLogger.Debug().Msg("Started launch instance AWS job")
-
 	args := LaunchInstanceAWSTaskArgs{}
 	err := job.Decode(&args)
 	if err != nil {
-		ctxLogger.Error().Err(err).Msg("unable to decode arguments")
+		ctxval.Logger(ctx).Error().Err(err).Msg("unable to decode arguments")
 		return fmt.Errorf("unable to decode args: %w", err)
 	}
+	logger := ctxval.Logger(ctx).With().Int64("reservation_id", args.ReservationID).Logger()
+	logger.Info().Interface("args", args).Msg("Processing no operation job")
```

Comment:
I think this msg should be regarding Launch Instance Job

Reply:
Good catch, you know what? I see a lot of copy&pasted code in these handlers, so I created a bunch of helper functions in that package which shortens this to just:

```go
// Unmarshall arguments and handle error
func HandleNoop(ctx context.Context, job dejq.Job) error {
	args := NoopJobArgs{}
	decodeJob(ctx, job, args)
	ctx = contextLogger(ctx, job.Type(), args, args.AccountID, args.ReservationID)

	jobErr := handleNoop(ctx, &args)

	finishJob(ctx, args.ReservationID, jobErr)
	return jobErr
}

// Job logic, when error is returned the job status is updated accordingly
func handleNoop(ctx context.Context, args *NoopJobArgs) error {
        // ...
}
```